src/main/java/org/example/LessonDao.java

```diff
@@ -14,21 +14,15 @@ public void addLesson(Lesson lesson) {
              PreparedStatement statement = connection.prepareStatement(query)) {
 
             statement.setString(nameIndex, lesson.getName());
-
-            if (lesson.getHomework() != null) {
-                statement.setInt(homeworkIdIndex, lesson.getHomework().getId());
-            } else {
-                statement.setNull(homeworkIdIndex, java.sql.Types.INTEGER);
-            }
+            statement.setObject(homeworkIdIndex, (lesson.getHomework() != null) ? lesson.getHomework().getId() : null);
```

Comment:
Almost, but still not correct. The type of homework id is `int`. Why are you using `statement.setObject()`? You should use `statement.setInt()` instead.